### PR TITLE
GHV: Adds Error handling to ProfileSearch.

### PR DESCRIPTION
**Pull request by DCox2016**
Adds message component, http-error-handler-service, and catchError to profile-search.service.ts

### CODE REVIEW

**Comment by cpyle0819** [comment]
PR title should match Mob standards.

File: src/app/http-error-handler.service.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
```
Comment (cpyle0819) [inline src/app/http-error-handler.service.ts]: I think this error handling service is overkill for your app. If it's working, I don't see anything wrong with the concept per se, it's just a lot of logic to handle errors from one source.

File: src/app/profile-show/profile-show.component.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+@Component({
+  selector: 'ghv-profile-show',
+  templateUrl: './profile-show.html',
+  styleUrls: ['./profile-show.css']
+})
+
+
+export class ShowProfileComponent {
+  profile$ = this.searchService.profile;
```
Comment (cpyle0819) [inline src/app/profile-show/profile-show.component.ts]: In this component you're naming your observables with a `$`, but in the search service you're not. It's best to pick a way and stick with it.

File: src/app/profile-show/profile-show.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<div class="profileInfo">
+    <div class="userInfo">
+        <h2>User's name: {{ (profile$ | async)?.name }}</h2>
```
Comment (cpyle0819) [inline src/app/profile-show/profile-show.html]: I don't think you need the `?` operator on these any more. You would only need that if `'profile$'` might not exist.

File: src/app/http-error-handler.service.ts (hunk 1 above)
Comment (DCox2016) [reply src/app/http-error-handler.service.ts]: fixed

File: src/app/profile-show/profile-show.component.ts (hunk 2 above)
Comment (DCox2016) [reply src/app/profile-show/profile-show.component.ts]: fixed

File: src/app/profile-show/profile-show.html (hunk 3 above)
Comment (DCox2016) [reply src/app/profile-show/profile-show.html]: fixed

**Pull request merged**